scripts/sct_utils.py

```diff
@@ -401,19 +401,19 @@ def my_emit(self, record):
 #=======================================================================================================================
 def add_suffix(fname, suffix):
     """
-    Add suffix between end of file name and extension on a nii or nii.gz file.
+    Add suffix between end of file name and extension. If the file name is pouf.nii.gz, then the suffix will be added
+    before the .nii.gz. Like that: poufsuffix.nii.gz
     :param fname: absolute or relative file name. Example: t2.nii
     :param suffix: suffix. Example: _mean
     :return: file name with suffix. Example: t2_mean.nii
     """
-    # get index of extension. Here, we search from the end to avoid issue with folders that have ".nii" in their name.
-    ind_nii = fname.rfind('.nii')
-    # in case no extension was found (i.e. only prefix was specified by user)
-    if ind_nii == -1:
-        return fname[:len(fname)] + suffix
+    # check if extension is .nii.gz
+    if fname[-7:] == '.nii.gz':
+        # get index of extension
+        ind_ext = fname.find('.nii.gz')
```

Comment:
Doesn't work for ``pouetpouet.nii.gz-pouet.nii.gz``, the previous ``.rfind()`` did.
Also, you could have done ``fname[:-7]``.
But maybe you could just piggy-back on ``sct.extract_fname()`` which makes it even cleaner.
Which made me realized that its code is messy.
I imagine that you have better things to do, I can clean that up later.